src/lib/utils.ts

```diff
@@ -429,3 +435,58 @@ export function hashData(
 
   return hashFunction.method(data);
 }
+
+/**
+ * Hashes the data received with the specified hashing function,
+ * and compares the result with the provided hash.
+ *
+ * @throws *Error* in case of a mismatch of the hashes.
+ */
+export function hashAndCompare(
+  data,
+  hash: string,
```

Comment:
what about renaming it to `expectedHash` ?
So later in the function when we read `hash` we know which hash it is.